src/main/java/org/zalando/intellij/swagger/completion/value/completion/openapi/ComponentRefValueCompletion.java

```diff
@@ -24,11 +31,37 @@
 
   @Override
   public void fill() {
-    getSchemaKeys().forEach(this::addValue);
+    final PsiFile psiFile = completionHelper.getPsiFile().getOriginalFile();
+    final Project project = psiFile.getProject();
+    final PsiManager psiManager = PsiManager.getInstance(project);
+    final VirtualFile virtualFile = psiFile.getVirtualFile();
+
+    getSchemaKeys(psiFile).forEach(this::addValue);
+
+    if (virtualFile != null && virtualFile.getExtension() != null) {
+      final Module module = ProjectFileIndex.getInstance(project).getModuleForFile(virtualFile);
+
+      if (module != null) {
+        FilenameIndex.getAllFilesByExt(
+                project, virtualFile.getExtension(), GlobalSearchScope.moduleScope(module))
+            .forEach(
+                f -> {
+                  PsiFile file = psiManager.findFile(f);
+                  if (file != null && !psiFile.equals(file)) {
+                    getSchemaKeys(file)
+                        .stream()
+                        .map(
+                            v -> {
+                              return new StringValue("./" + file.getName() + v.getValue());
```

Comment:
Using relative paths has really difficult to predict consequences on how a reference is behaving, when the file is served by a server or resolved within an application. I definitely would prefer absolute paths, so that applications do not need to track the origin of files, however as long as this feature is not able to create full qualified `URLs`, the gain of absolute paths is minor and my in rare occasions have a negative impact.

Reply:
I will look into whether we can implement this like the SMART completions in the IJ. So it will not affect the normal completions.

About the relative path "./", maybe we can get rid of it, So that that actual generator can decide how to resolve the absolute path for the file.

WDYT @tkrop ?

Reply:
Yes, sounds better.